app/models/course.rb

```diff
@@ -304,4 +305,11 @@ def ensure_required_params
     self.timeline_start ||= start
     self.timeline_end ||= self.end
   end
+
+  def set_end_times
+    unless self.use_start_and_end_times
+      self.end = self.end.end_of_day
+      self.timeline_end = self.timeline_end.end_of_day
```

Comment:
This method should also set the start times to 00:00. That will ensure consistency if a course changes type from one that uses start/end times to one that doesn't.